### PR TITLE
chore(webpack-cli): remove --no-mode flag

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**

Follow up #1481 .
remove `--no-mode`.

**Did you add tests for your changes?**
Removed `no-mode` tests.

**If relevant, did you update the documentation?**
Yes

**Summary**

removed `--no-mode` because it is same as `--mode none` and Negated flags makes more sense with boolean args.

**Does this PR introduce a breaking change?**
Maybe

**Other information**

### CODE REVIEW

File: packages/webpack-cli/lib/groups/ZeroConfigGroup.js [hunk 1]
```diff
@@ -20,18 +20,14 @@ class ZeroConfigGroup extends GroupHelper {
...
-            if ((this.args.mode || noMode) && (this.args.dev || this.args.prod)) {
+            if (this.args.mode && (this.args.dev || this.args.prod)) {
                 logger.warn(
-                    `You provided both ${this.args.mode ? 'mode' : 'no-mode'} and ${
+                    `You provided both 'mode' and ${
                         this.args.prod ? '--prod' : '--dev'
-                    } arguments. You should provide just one. "${this.args.mode ? 'mode' : 'no-mode'}" will be used`,
+                    } arguments. You should provide just one. "mode" will be used`,
                 );
                 if (this.args.mode) {
```
Comment (Loonride) [inline packages/webpack-cli/lib/groups/ZeroConfigGroup.js:29]: small nitpick, we don't need this if statement here anymore

Comment (snitin315) [reply packages/webpack-cli/lib/groups/ZeroConfigGroup.js]: @Loonride Done :+1:

**Comment by evilebottnawi** [comment]
Thanks!

**Pull request merged**